src/pages/User/Learning/TodaysWord/TodaysWord.tsx

```diff
@@ -0,0 +1,127 @@
+import React, { useContext, useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { toast } from 'react-toastify';
+import fetchWithToken from 'API/api';
+import * as S from './StyleTodaysWord';
+import { Button } from 'components/Shared/Buttons/Button';
+import NotificationInformation from 'components/Shared/Atoms/NotificationInformation';
+import ManagedIcon from 'assets/icon/helpers/ManagedIcon';
+import { GlobalContext } from 'utils/GlobalContext';
+
+const TodayWord = () => {
+  const { isLoginUser } = useContext(GlobalContext);
+  const [baseWord, setBaseWord] = useState(null);
+  const [transWord, setTransWord] = useState(null);
+  const [loading, setLoading] = useState(true);
+  const [isError, setIsError] = useState(false);
+  const [isShowWord, setIsShowWord] = useState(false);
+  const [isSpeaking, setIsSpeaking] = useState(false);
+  const [idWord, setIdWord] = useState(null);
+
+  const navigate = useNavigate();
+  const handleAddWordClick = () => {
+    navigate('/settings/add_words');
+  };
+
+  const handleShowWord = () => {
+    setIsShowWord(!isShowWord);
+  };
+
+  const handleSpeak = () => {
+    if (transWord) {
+      setIsSpeaking(true);
+      const utterance = new SpeechSynthesisUtterance(transWord);
+      utterance.lang = 'en-US';
+      utterance.onend = () => {
+        setIsSpeaking(false);
+      };
+      window.speechSynthesis.speak(utterance);
+    }
+  };
+
+  const handleLearnedWord = async () => {
+    if (idWord === null) {
+      toast.error('Word ID is missing.');
+      return;
+    }
+    try {
+      const { response, status } = await fetchWithToken({
+        endpoint: 'updateWord',
+        method: 'PUT',
+        body: { status: 2 },
+        params: idWord,
+      });
+
+      toast.success('Word status updated successfully!');
+      fetchData();
+    } catch (error) {
+      toast.error('Failed to update word status.');
+      console.error(error);
+    }
+  };
+
+  const fetchData = async () => {
+    setLoading(true);
+    try {
+      const { response, status } = await fetchWithToken({
+        endpoint: 'getTodayWord',
+      });
+      setBaseWord(response.basicWord);
+      setTransWord(response.transWord);
+      setIdWord(response._id);
+      setIsError(false);
+    } catch (err) {
+      toast.warn(
+        'Failed to fetch the word of the day as there are no new words to learn. Please add new ones.'
+      );
+      setIsError(true);
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    if (isLoginUser) {
```

Comment:
możesz to krótko zapisać: if (isLoginUser) fetchData();
masz wtedy 1 linijkę zamiast 3